src/components/multiselect/MultiSelect.vue

```diff
@@ -1063,7 +1063,23 @@ export default {
         visibleOptions() {
             const options = this.optionGroupLabel ? this.flatOptions(this.options) : this.options || [];
 
-            return this.filterValue ? FilterService.filter(options, this.searchFields, this.filterValue, this.filterMatchMode, this.filterLocale) : options;
+            if (!this.filterValue) return options;
+
+            const filteredOptions = FilterService.filter(options, this.searchFields, this.filterValue, this.filterMatchMode, this.filterLocale);
+
+            if (!this.optionGroupLabel) return filteredOptions;
+
+            const optionGroups = this.options || [];
+
+            const filtered = [];
+
+            optionGroups.forEach((group) => {
+                const filteredItems = group.items.filter((x) => filteredOptions.includes(x));
```

Comment:
```suggestion
                const filteredItems = group.items.filter((item) => filteredOptions.includes(item));
```
Simple change. Thank you for your contribution.

Reply:
Updated variable name as suggested